website/src/content/docs/docs/howtos/Generate client libraries/02client.mdx

```diff
@@ -555,3 +555,75 @@ petActionClient.pet();
 ```
 
 </ClientTabs>
+
+### Adding Client Initialization Parameters
+
+By default, we only generate our clients with initialization parameters for `endpoint`, `credential`, and `apiVersion`, whenever any of these are applicable. There are cases where spec authors would like their clients to have additional input parameters. This is where the `@clientInitialization` decorator comes in.
+
+With `@clientInitialization`, you can pass in additional parameters you would like your client to have, by passing in an input model. All properties of the input model will be appended to the current default list of client initialization parameters. Additionally, these client parameters will no longer appear on service methods that previously had them as part of the method signature. The generated code will automatically pass in the inputted value from the client init to the service.
+
+<ClientTabs>
+
+```typespec title="main.tsp"
+@service
+namespace Storage;
+
+op upload(blobName: string): void;
+op download(blobName: string): void;
```

Comment:
Can we use `@path` as an example (e.g. `@path blobName: string)`)?

There is currently limitation in Java emitter when this "option to clientInitialization" is a property within a request body (instead of a normal parameter like path/query/header).

Reply:
ok I'll update the docs